### ISSUE TITLE
Early Incentivization for Validators

### ISSUE THREAD

**Issue by rigelrozanski** [cosmos/cosmos-economics#26]
# Issue

If fees are very low, as they may be in the early days of cosmos, validators are not incentived to validator if we do not allow them to charge commission on the validator provisions of atoms

# Slack Discussion 
**crainbf**
That’s insane. Of course, validators should be able to charge commission on that too
There could be two separate commission rates validators can set so if some  only want to charge on tx fees, fine
But prohibiting charging commission on block rewards will make this economically totally infeasible
And I’m not just super opposed to this in the short run, but in the long run too
gamarin where does this idea come from? What is the thinking behind it?

**gamarin**
I think rige could explain it better than me, but basically the idea is to prevent a continuous shift in distribution from delegators to validators. Atom are already quite concentrated, and at  20% inflation rate with  commissions validators would quickly end up in possession of a significant proportion of total Atoms

**crainbf**
I’m really concerned in what I perceive is this central planner mindset. This is a perfect expression of it
I think we need to give as many options as possible both to delegators and validators
They should be able to enter whatever economic arrangement they want. We should allow actual markets to form where people can have different agreements
It seems there is a big focus on taking options away because of some idea for what kind of agreements parties should enter.

**crainbf**
I’m also convinced that running a validator is economically infeasible if commission cannot be charged on block rewards.
 3 replies Last reply today at 9:21 PM View thread

**crainbf**
But more importantly, if a validator wants to charge commission on block reward and delegators are happy to delegate to them. Who are you guys to say: This isn’t allowed! Seems completely wrong to me
If one can run validators profitably just charging commission on transaction fees, obviously the validators that do that have a more competitive offer and should gain more delegators.
So: more fine-grained options for charging fees so that validators can choose to only charge on tx-fees => great idea
Taking option away completely to charge on block rewards => awful idea

**gamarin**
I don't think anything is absolutely set in stone yet. Now FAQ is going to be released and feedback is welcome. I personally believe that it's healthy to compare alternatives so that we end up choosing the best one.
So Brian proposes: "Have two different commissions, one for fee and one for block provisions. Let validators set these two commissions freely". What's your take on it rige adrian sunnya97? (edited)

**crainbf**
channel
My thoughts on the ability of validators to charge commission on block rewards: https://www.reddit.com/r/cosmosnetwork/comments/7hho47/why_validators_need_to_be_able_to_charge/
I think this issue is of critical importance and the current proposal would dramatically impair our chance of succeeding

**jimyang**
You want to cross post this into our forum?

**crainbf**
jimyang Good point. Will do

**zaki**
Endorse crainbf ‘s view here
Mentioned on reddit

**ethan.frey**
I agree also

**rige**
In the early days of the network - I don’t think it’s a problem if the fees commission needs to be 100% (as delegators still get block provisions)… the network is really NOT supposed to run on provisions, provisions are only there to encourage bonding. If a validator is unable to sufficiently run on fees I’m not sure they should be running a validator

**rige**
Like check it: The security of the system should only really reflect the worth of the system - we don’t need insane security if what we are securing has low value, as low value is low incentive to break in. As such I really don’t want to see increased security based on revenues from selling atom provisions - It’s not meant to be a bootstrapping mechanism is it? - like just unbond and sell some of your atoms if you need to bootstrap?
I think we need a strong force able to access the security of each validator and rapidly shift that power if necessarily - if we allow validators to charge commission on this they we create degrade that force (to nil eventually)

**rige**
Although this is still highly contentious for me - I think there is no argument against block-provision fees to aid bootstrapping the network. It’s just a question of whether we want the initial distribution to shift (towards the validators) during the formative days of the network or whether we are okay with excluding some validators and likely sacrificing security in this early phase… If our ultimate mission is to bootstrap the network, artificial rewards on the provisions make sense - and the network propose to remove them through governance once the network had matured

**gamarin**
I think Brian's double commission mechanism allows us to naturally shift towards equilibrium. If commission on provision do not make sense anymore, delegators will switch to validators that apply low to zero commission on block provision.
Even if these validators' commission on fee are higher

**bucky**
doubel commision? ie lower commission on block provisions, higher commission on fees?

**gamarin**
Double commission: Validators can set different commission on fees and block provisions

**bucky**
yeh. with the intention/hope that theyd do lower commision on provisions

**bucky**
can we enforce that ?
1 reply Today at 5:11 PM View thread

**gamarin**
So one validator could choose to apply 0 commission on block provision and 50% commission on fees
I mean the power is in the hands of delegators. If they value more validators that apply low commission on block provision, this should become the norm

**bucky**
im worried the power might be too much in the hands of the validators
and they could just set the commissions really high
whats a delegator to do ?

**gamarin**
But it's a competitive market right ?
If validator set arbitrary high commissions, delegators will switch
We should make sure markets can function efficiently
By smoothening the switching process

**bucky**
yeh thats the idea
so we should have two commission rates, totally independent of eachother ?

**gamarin**
that's Brian idea yeah. Each commission is independent from eachother.
Also we have to remember that a validator's commission on provision actually affect its delegators the most. If it's too high and redistribution is unbearable delegators will switch. Even if commission on fee is low.

**rige**
We could certainly get it to work - two commissions is a bit complicated to digest but whatever. My two main criticism are:
 1) shift of power, which weakens the systems ability to jump from bad validators
 2) security of the system won’t reflect the value of the system (overly conservative)

**crainbf**

**rige**
`2) security of the system won't reflect the value of the system (overly conservative)`
This is inaccurate. Block rewards and validators being able to charge commission on block rewards are exactly what will get the security of the system in line with the value of the system.
Think of it like this: Let’s say Cosmos market cap is $500m, but there are just a few hundred transactions a day. Primarily people moving coins to and form exchanges. So revenues of validators are basically zero in that context. There is absolutely no connection between validator revenues and market cap. 
Block reward and commission on that is what creates that link since it will increase proportionally with market cap.
`If a validator is unable to sufficiently run on fees I'm not sure they should be running a validator`
<- The point of my reddit post was exactly why the dynamics fo the system will be such that validators will not be able to generate significant revenues from tx fees. This is the case at least until we switch to a mode with full blocks like it is happening in Bitcoin.

**rige**
intestesting - so what you’re saying is we effectively limit our own market cap by having reduced incentive for security

**crainbf**
Not exactly. Our market cap could still be high. It can be totally decoupled from the underlying security infrastructure, but that makes for a very fragile network

**rige**
yes of course
(rational market only)

**crainbf**
Imagine a $1bn network that is run by people having nodes in their basement, because you can’t generate revenues from running a miner so nobody does it as a professional business

**rige**
Well - Maybe the market shouldn’t reach that high until there is a good reason for it to reach that high (aka transactions!)… and also - if you were a validator, the worth of your staked atoms would be huge (as your portion of staked atoms is always increasing)… so why would you need to be charging commission on provisions?  - I think your point on the bootstrap problem is relevant depending on how we want the system to look in the early formative days of usage - almost convinced that it’s a good idea then - but I’m really not convinced that it makes sense in the long term to be charging on validator provisions (please note my language, provision, not reward, two totally different ideas) (edited)
This question comes down to how to want to roll out this network as well… do we want a big (and potentially insecure) bang, followed by potential hacks? or do we want a slow, secure rollout of the network… I’m convinced it’s the latter (not that there aren’t more options of course) (edited)

**crainbf**
And I’m convinced your path is the path to an insecure roll-out
Why would it be secure if there is no money to be made so nobody invests in running secure validators? That makes zero sense

**rige**
^ hey I’m mostly convinced that what you’re saying makes sense for bootstrapping

**crainbf**
Take us. We’re prepared to invest a very substantial amount of money to run a very high quality validator. But frankly, if one can’t charge commission on the block reward, it’s a guaranteed-money losing proposition. So we may decide not to do it. Others will do the same

**rige**
just not for long term operation

**crainbf**
Rigel, you’re not the central planner of Cosmos

**rige**
neither are you - it would have to be governance that would make a change

**crainbf**
Sure. Totally aligned on that
Here is what the default should be:
Have the maximum number of options available as the default. If options will be removed, it should be through governance votes

**rige**
I disagree with that as a general statement - we are making all sorts of limiting calls I reckon, beyond this specific issue (edited)

**crainbf**
I think limiting calls are okay, but only if there are strong reasons for it. These could be technical complexity or some other strong reason

**rige**
yeah I think there is a really good reason to limit commission being charged on provisions as it will weaken the longterm security of the system through centralization of the atom holders. I think you’ve raised a great point about problematic inventive structure during the early days of the network, provision commission may be the only solution - however I wish there was another way, and I’m going to think more about it. (edited)

**crainbf**
Okay. If you object to my arguments, can you please write up a proper/in-depth answer as a response on reddit and the forum?
Explain where they are wrong
We should have this discussion in a structured way in public

**rige**
I need to focus on programming for right now, this is still an open issue I think - I’m going open up a github issue in `cosmos/cosmos-economics`

**Comment by rigelrozanski** [comment]
Here is a link to the reddit post https://www.reddit.com/r/cosmosnetwork/comments/7hho47/why_validators_need_to_be_able_to_charge/

**Comment by rigelrozanski** [comment]
Possible solution:
Allow validators to charge commission on provisions - propose governance to change this at some future date once fees are high enough

**Comment by crainbf** [comment]
I'm fine with the possible solution. Two different fee rates are great. This will allow validator to charge no fees on block rewards if they choose so.

And, of course, if atom holders later choose to have a governance change and remove that ability for everyone, then that's their right to do so. At that point, there will also be objective data and a basis for the discussion.

**Comment by rigelrozanski** [comment]
Also, I'd propose that if you're going to charge commission on the atom provisions - you must first be charging 100% commission on the fees first! Then maybe we really don't need this to switch ever this could be a continuous threshold... if validators can be operating on the fees alone - this is the way which this is achieved

**Comment by crainbf** [comment]
I disagree with this. I think it is fundamentally wrong for us to dictate how people run their validators. It will lead to less diversity and a weaker network.

There should be no restrictions on how validators charge fees. You can later do a governance vote for this and then people can decide that if they want it.

**Comment by rigelrozanski** [comment]
Okay - I think we can leave this as the fundamental disagreement - I think we should dictate this as the basis as it will lead to a more diverse (less centralized to the validators) set of atom holders which will strengthen the system by allowing delegators to quickly react and move power to/from validators depending on circumstance

**Comment by adrianbrink** [comment]
I agree with @rigelrozanski on this one. We are and should be opinionated on Cosmos as a network. Also, I think we should design the protocol to incentivise a validator set which is as diverse as possible. 

Imagine if Ethereum had always wanted to be non-opinionated. They would have never built the current network.

**Comment by gamarin2** [comment]
I think this is a textbook example of a fatal conceit. We are navigating uncharted waters here, and we cannot possibly imagine all the consequences of an artificial constraint like forcing validators to impose 100% commission on fees before being able to apply a commission on provisions.

- You are saying that forcing validators to apply 100% commission on fees before being able to apply a commission on provisions will lead to a more diverse set of Atom holders. First, this might not even be true. For example, if we apply this constraint, it will be very hard for validators with a low self-bond ratio but amazing service to increase their proportion of Atoms. OTOH, validators with poor performance but high self-bond ratio will maintain their superiority. If we allowed the two commissions to fluctuate freely, we may see situations where delegators accept to delegate to a highly performant validators with low self-bond and 5% commission on provisions but do not accept to delegate to an average validators with high self-bond and 5% commission on provisions, because the latter validator already concentrates many Atoms in its hands. In other words, the market could tend to redistribute Atoms to high-performance validators with low self-bond and impose validators with high self-bond to have 0% commission on provisions.
- Even if it were true that the constraint would lead to a more diverse set of Atom holders (but certainly not a more diverse set of validators), it still disregards all the possible consequences on the network. Maybe this rule will lead to very high fees. Maybe it will drive away Atom holders from the network, because fees are paid in multiple currencies while provisions are paid in Atoms. If it's not profitable for validators to have less than 100% commission on fees, you effectively forbid delegators to collect any fee in any currency. Overall, it could lead to a multitude of bad scenario that we cannot foresee right now. 
- Thus, we need to go with the safest option first. And the safest in this case is to leave as many options open for both delegators and validators to find market optimum. Maybe delegators will find that 50% average commission on fees and 2% average commission on provisions is the best for the network to function in an optimal way. Saying that 100% commission on fees should be applied before commission on provision can be applied means that you think you know better how the market should function than the entirety of delegators and validators combined. 

In conclusion, we are all opinionated on how the network should function. But we should resist the urge to profit from our position of power and acknowledge that we might be wrong. This should lead us to design a network where we leave as many options open as possible for all parties, so that the market can find its natural optimum. 
Instead of constraining commissions, we should ensure that delegators can switch between validators smoothly, so that they can react and iteratively find, for each validator, the best values for both commissions.

**Comment by rigelrozanski** [comment]
> For example, if we apply this constraint, it will be very hard for validators with a low self-bond ratio but amazing service to increase their proportion of Atoms

Not the case, they can buy more atoms with all the commission on fees they are making on their huge delegator base b/c they're so great

> OTOH, validators with poor performance but high self-bond ratio will maintain their superiority

This is true in either case

>If we allowed the two commissions to fluctuate freely, we may see situations where delegators accept to delegate to a highly performant validator with low self-bond and 5% commission on provisions but do not accept to delegate to an average validator with high self-bond and 5% commission on provisions, because the latter validator already concentrates many Atoms in its hands

This seems rational but there is no way to know...  I think fundamentally is a validator needs to have a higher self bond ratio they can just charge more commission on fees and then buy those atoms at market price. Tangent: I don't think we want to be designing cosmos for delegators or validators who are NOT atom maximalists - this is kind of a fundamental point - we should probably design this system assuming all the parties always want more atoms. 

>Even if it were true that the constraint would lead to a more diverse set of Atom holders (but certainly not a more diverse set of validators)

Not certain this is true (it may be) - if we have a greater distribution of atom holders it may lead to more opportunities for those who would normally be delegators to become validators. 

> If it's not profitable for validators to have less than 100% commission on fees, you effectively forbid delegators to collect any fee in any currency. Overall, it could lead to a multitude of bad scenarios that we cannot foresee right now.

I don't think it's a problem, delegators will still be collecting more atoms! - even if they were not able to get any fees.... Just think about almost EVERY other blockchain... if you are a part of mining pool, which currency to you end up collecting... the native currency = not a problem for delegators. And I think it makes sense to design the system around atom maximalists... if a delegator in the end wants some other token, they can go to the market and trade for it. 

> Saying that 100% commission on fees should be applied before commission on provision can be applied means that you think you know better how the market should function than the entirety of delegators and validators combined

First of all, please refrain from using accusatory language - these are just ideas, I don't own them, and I'm not attached to them, just trying to navigate to the best solution, and I'm skeptical, have not been convinced otherwise, but I'm very open too it. 

In essence, adding this constraint is going to limit certain types participants to the network, and maybe also allow for new types of participants - I think it's quite possible that if validators are able to charge commission on provisions without having to change 100% commission on fees first, they ALL might charge commission on provisions, which would effectively limit participants who knew they were just being weened out of the system. Additionally I'm not sure we are really so desperate that we want every permutation of validator to be joining this network - maybe it's okay to be excluding validators who absolutely not live with a more diverse distribution. 
____________________________________________________
So in conclusion, I'm spent engaging on this stuff, I'm not emotionally attached to whatever decision we make, but I'm just about ready to move on, and I don't think I've been convinced that there is a good reason to not implement the mechanism I've described - hoping that Jae/Bucky can weigh in soon with further comment/decision

But good conversations folks! (if you don't hear from me much more on this one)

**Comment by gamarin2** [comment]
> Not the case, they can buy more atoms with all the commission on fees they are making on their huge delegator base b/c they're so great

Ok fair, but there is added friction in the process, because they have to go on exchanges which costs time, effort and fees.

> This is true in either case

I disagree. As I said, there is a possibility that delegators will accept commission on provisions for validators with low self-bond but not for validators with high self-bond. This would decrease the superiority of high self-bond validators compared to low self-bond validators over time. Admittedly this does not change the high self-bond validator's share of the network, but it allows delegators to voluntarily dilute themselves in order to increase a low self-bond validator's power.

> I don't think it's a problem, delegators will still be collecting more atoms! - even if they were not able to get any fees.... Just think about almost EVERY other blockchain... if you are a part of mining pool, which currency to you end up collecting... the native currency = not a problem for delegators. And I think it makes sense to design the system around atom maximalists... if a delegator in the end wants some other token, they can go to the market and trade for it.

I don't think it makes sense to consider the situation only from an Atom-maximalist standpoint. The value of Atoms is derived from the expected revenue from future transaction fees. Again, if, as a delegator, you can't directly collect your share of fees, you need additional steps to collect them, which adds friction. And this friction is only added for delegators. It translates to real costs in time, effort and money, which could make the network less attractive.

> First of all, please refrain from using accusatory language - these are just ideas, I don't own them, and I'm not attached to them, just trying to navigate to the best solution, and I'm skeptical, have not been convinced otherwise, but I'm very open too it.

Please don't take it personally. I was just making a logic statement "Saying X implies Y", this was not directed to you as an individual. What I was trying to say is that it does not make sense to me that a constrained market would be more efficient in finding network optimum than a less constrained one. 

> In essence, adding this constraint is going to limit certain types participants to the network, and maybe also allow for new types of participants - I think it's quite possible that if validators are able to charge commission on provisions without having to change 100% commission on fees first, they ALL might charge commission on provisions, which would effectively limit participants who knew they were just being weened out of the system. Additionally I'm not sure we are really so desperate that we want every permutation of validator to be joining this network - maybe it's okay to be excluding validators who absolutely not live with a more diverse distribution.

I don't think we would be limiting types of participants, we would just be limiting the set of options they can choose from, and I'm uncertain this would lead to a better outcome for the network. 
If all validators decide to charge commission on provisions, then it either means that they can be profitable without it OR that they have the full consent of delegators. If validators can be profitable without commission on provisions and delegators do not want them to charge commission on provisions, then delegators will keep switching to validators that charge lower and lower commission on provisions until it reaches 0. This is an open market: if there is profit to be made and market share to be taken, new validators will pop up with more competitive commission on provisions. 

I really think this is important we reach consensus on that. I must admit that I am still failing to see what the benefits of imposing 100% commission on fees before commission on provision can be applied are.

**Comment by zmanian** [comment]
A thought I had recently.

The validator provision is effectively a wealth transfer from traders/HODLers to delegators and validators.

The validator that is enabling this should be able to charge rent on this wealth transfer as part of the system.

**Comment by rigelrozanski** [comment]
yeah maybe, I think there is some kind of consensus on that - especially for early days - the real question is do we want to disincentive it over changing commission for the fees, aka as a validator, if you can get the commission you need from only the fees then you should, as to not drive centralization in the atom distribution

**Comment by zmanian** [comment]
I see incentive to strongly motivate competition on the commission validators gain from staking atoms

**Comment by rigelrozanski** [comment]
Outcomes based on conversation with @jaekwon: 
Also see Jae's reddit response: https://www.reddit.com/r/cosmosnetwork/comments/7hho47/why_validators_need_to_be_able_to_charge/
 - For simplicity let's use 1 commission rate applied for both fees and provisions (also photon provisions which we haven't talked about much)
 - For technical reasons, atom provisions need to be applied as auto-bonded (illiquid) therefor this commission will be applied in this illiquid setting (slightly more complicated math)
 - Governance can modify this to something more specific post launch if there is an appetite for it

**Comment by zmanian** [comment]
ACK

**Comment by sunnya97** [comment]
Since we will have Photon rewards from Day 1, validators can just take commission from that.  That way, even when the network is young and there's not a lot of tx fees, validators can still take a large cut from Photons if they want.  Putting commission on Atom provisions is undesirable because over time all the staking power will flow from delegators to validators.

**Comment by rigelrozanski** [comment]
Yeah photons kind of change the conversation, If it is likely financially feasible for validators to be able to sustain on photons and fees then I second Sunny's comment. We don't need to charge commission on atoms. In the early days of the network, even if there are low fees, photon provisions should still be existent... This is the alternative I was hoping existed. 

Second note @sunnya97 do we have a Creation schedule for the Photons? I was thinking this makes sense to be a mining-like schedule.. Thoughts?

**Comment by gamarin2** [comment]
Bucky said Ethermint might not be ready for launch (even as hard spoon zone) before April. So if Hub launches in Feb and Spoon Zone in April, that's two months without photons.

**Comment by jaekwon** [comment]
> Putting commission on Atom provisions is undesirable because over time all the staking power will flow from delegators to validators.

If you do the math, it actually takes longer than you'd think.  That said, there's a case to making staking accessible to a broader community of keepers than only the validators, and for that the long-term does matter.

> We have to consider the possibility that Ethermint won't be ready when Hub launches

That's OK, we can still inflate photons before the hard-spoon occurs.

**Comment by jaekwon** [comment]
Originally I had thought that the latter (e.g. eventually yielding to the validators) but I think for the Cosmos Hub it makes more sense to have a balance of delegators.

I'm also not sure that this actually matters much if there magically exists a liquid market of conversion between atoms and photons.  If you assume that the price of atoms and photons and the % of bonded atom are relatively stable, then maybe we can conclude that the two choices are more or less the same after conversions and adjustments to commissions.

**Comment by rigelrozanski** [comment]
@jaekwon 
>Originally I had thought that the latter (e.g. eventually yielding to the validators) but I think for the Cosmos Hub it makes more sense to have a balance of delegators.

Agreed, I think having a powerful set of delegators will lead to a more diverse cosmos community, and diverse groups generally make better decisions = applicable for governance. Not to mention validators who want to hedge. Not to mention the validator watcher role which delegators can hold, to flock away from crappy/malicious validators, and towards new innovative delegators. 

> I'm also not sure that this actually matters much if there magically exists a liquid market of conversion between atoms and photons. If you assume that the price of atoms and photons and the % of bonded atom are relatively stable, then maybe we can conclude that the two choices are more or less the same after conversions and adjustments to commissions

I think there is an important distinction between buying the atoms from the market and receiving them, it's about having the upper hand in a market negotiation. This becomes increasingly important with larger volumes... if all delegators needed to constantly buy atoms to keep the same portion of the total atom supply (and wanted too) there would be large buy pressure on the market which would pump the price of atoms allowing the validators to double down on their profits. I'm all for validators expanding their operations and buying more atoms, I just don't reckon they should automatically have the upper hand in this negotiation.
